src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -400,7 +400,16 @@ private static void CheckPathValidity(string path)
         /// </summary>
         private static bool IsEffectivelyEmpty(ReadOnlySpan<char> path)
         {
-            return (path.IsEmpty || path.IndexOf(' ') > -1);
+            if (path.IsEmpty)
+                return true;
+
+            foreach (char c in path)
+            {
+                if (c != ' ')
+                    return false;
+            }
+
```

Comment:
`return path.IsWhiteSpace();` ?  (it returns true if ROS<char> is empty or contains only spaces)